From d45cc8df7f59eb4db28408076ce979cd5e18f2b7 Mon Sep 17 00:00:00 2001
From: Johan Hovold <jhovold@gmail.com>
Date: Sat, 8 May 2010 15:18:41 +0200
Subject: USB: pl2303: use tty_insert_flip_string_fixed_flag
Patch-mainline: v2.6.35
Git-commit: d45cc8df7f59eb4db28408076ce979cd5e18f2b7

Use tty_insert_flip_string_fixed_flag to report errors to line
discipline.

Signed-off-by: Johan Hovold <jhovold@gmail.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/serial/pl2303.c |    7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

--- a/drivers/usb/serial/pl2303.c
+++ b/drivers/usb/serial/pl2303.c
@@ -800,12 +800,13 @@ static void pl2303_process_read_urb(stru
 	if (line_status & UART_OVERRUN_ERROR)
 		tty_insert_flip_char(tty, 0, TTY_OVERRUN);
 
-	if (tty_flag == TTY_NORMAL && !(port->port.console && port->sysrq))
-		tty_insert_flip_string(tty, data, urb->actual_length);
-	else {
+	if (port->port.console && port->sysrq) {
 		for (i = 0; i < urb->actual_length; ++i)
 			if (!usb_serial_handle_sysrq_char(tty, port, data[i]))
 				tty_insert_flip_char(tty, data[i], tty_flag);
+	} else {
+		tty_insert_flip_string_fixed_flag(tty, data, tty_flag,
+							urb->actual_length);
 	}
 	tty_flip_buffer_push(tty);
 	tty_kref_put(tty);
